src/backend/distributed/utils/colocation_utils.c

```diff
@@ -96,6 +98,74 @@ mark_tables_colocated(PG_FUNCTION_ARGS)
 }
 
 
+/*
+ * update_distributed_table_colocation updates the colocation of a table.
+ * if colocate_with -> 'none' then the table is assigned a new
+ * colocation group.
+ */
+Datum
+update_distributed_table_colocation(PG_FUNCTION_ARGS)
+{
+	Oid targetRelationId = PG_GETARG_OID(0);
+	text *colocateWithTableNameText = PG_GETARG_TEXT_P(1);
+
+	CheckCitusVersion(ERROR);
+	EnsureCoordinator();
+	EnsureTableOwner(targetRelationId);
+	EnsureHashDistributedTable(targetRelationId);
```

Comment:
Strictly speaking, range-distributed tables can also be co-located, so it would be nice if this function supported that. 

Since create_distributed_table only supports colocate_with for hash-distributed tables, I'm ok with leaving it as is.

Reply:
I was trying to be consistent with create_distributed_table. 

I think we should either have in both or none, so as you are also okay with leaving it as is I will keep it this way